kafka-connect/kafka-connect/src/main/java/org/apache/iceberg/connect/IcebergSinkTask.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.connect;
+
+import java.util.Collection;
+import java.util.Map;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.kafka.clients.consumer.OffsetAndMetadata;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.sink.SinkTask;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class IcebergSinkTask extends SinkTask {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IcebergSinkTask.class);
+
+  private IcebergSinkConfig config;
+  private Catalog catalog;
+  private Committer committer;
+
+  @Override
+  public String version() {
+    return IcebergSinkConfig.version();
+  }
+
+  @Override
+  public void start(Map<String, String> props) {
+    this.config = new IcebergSinkConfig(props);
+  }
+
+  @Override
+  public void open(Collection<TopicPartition> partitions) {
+    catalog = CatalogUtils.loadCatalog(config);
```

Comment:
`open` method is called with only the newly assigned partitions. Is there a strong reason to pass just the _newly assigned_ partitions to the `Committer.start` method when the Committer can just retrieve _all_ partitions assigned to this task via `context.assignment` anyway?

I'm also worried we might have a bug here. The Committer implementation uses this `partitions` argument to check if partition 0 of the first topic is assigned to this task and if so, it spawns a Coordinator process. I'm worried that if there was a rebalance where the partition 0 of the first topic doesn't move between tasks, then it would not be included in the `partitions` argument for any Task and thus we could potentially end up with a Connector that doesn't have any Coordinator process running on any Task. Thoughts?

Reply:
That's not my understanding, `open()` will be called with the new assignment, i.e. all assigned topic partitions. See the [javadoc](https://kafka.apache.org/37/javadoc/org/apache/kafka/connect/sink/SinkTask.html#open(java.util.Collection)): "The list of partitions that are now assigned to the task (may include partitions previously assigned to the task)"

Reply:
I did read the javadoc and interpreted it differently but I can totally see how you reached your interpretation as well. 

The reason I'm inclined towards my interpretation is because I know `SinkTask.open` is ultimately called by the `ConsumerRebalanceListener.onPartitionsAssigned` method ([here](https://github.com/apache/kafka/blob/958bc0601c7852b8aaf28960280b1f45efd7d850/connect/runtime/src/main/java/org/apache/kafka/connect/runtime/WorkerSinkTask.java#L741) in the Kafka source code) and if you look at the javadocs for that method, it says `The list of partitions that are now assigned to the consumer (previously owned partitions will NOT be included, i.e. this list will only include newly added partitions)`. 

I could still be wrong so feel free to test things out or ask in the kafka community.

Reply:
Thanks, it does seem like you're right. I changed this to get the consumer assignment instead.